compiler-rt/lib/builtins/aarch64/sme-abi-init.c

```diff
@@ -50,3 +52,24 @@ __attribute__((constructor(90)))
 static void init_aarch64_has_sme(void) {
   __aarch64_has_sme_and_tpidr2_el0 = has_sme();
 }
+
+#if __GNUC__ >= 9
+#pragma GCC diagnostic ignored "-Wprio-ctor-dtor"
+#endif
+__attribute__((constructor(90))) void get_aarch64_cpu_features(void) {
+  if (!__aarch64_cpu_features.features)
+    __init_cpu_features();
+}
+
+__attribute__((target("sve"))) long emit_cntd(void) {
+  long vl;
+  __asm__ __volatile__("cntd %0" : "=r"(vl));
+  return vl;
+}
+
+long get_runtime_vl(void) {
```

Comment:
This probably should be prefixed with `__aarch64`

Reply:
Hi @efriedma-quic, I've added this routine to the ABI document in https://github.com/ARM-software/abi-aa/pull/263 and renamed it to `__arm_get_current_vg`.